web/src/composebox_typeahead.ts

```diff
@@ -1338,7 +1338,55 @@ export function initialize({
     };
     new Typeahead(stream_message_typeahead_input, {
         source(): string[] {
-            return topics_seen_for(compose_state.stream_id());
+            const topics = topics_seen_for(compose_state.stream_id());
+            const topic_input = stream_message_typeahead_input.$element.val()?.toString() ?? "";
+
+            let dm_option: string[] = [];
+            if (topic_input.length >= 3) {
+                const matched_users = people
+                    .get_realm_users()
+                    .filter((user) =>
+                        user.full_name.toLowerCase().startsWith(topic_input.toLowerCase()),
+                    );
+
+                if (matched_users.length > 0) {
+                    dm_option = [`Switch to DM with ${matched_users[0]?.full_name}`];
+                }
+            }
+
+            return [...dm_option, ...topics];
+        },
+        updater(item: string): string | undefined {
+            if (item.startsWith("Switch to DM with")) {
```

Comment:
You can't construct or check strings this way in an internationalized product. See https://zulip.readthedocs.io/en/latest/translating/internationalization.html to learn how to do this properly. Likely you want to use a CSS class for this check.